### ISSUE TITLE
prefer-const: should ignore anything in a declaration

### ISSUE THREAD

**Issue by andy-ms** [palantir/tslint#2390]
### Bug Report

- __TSLint version__: `master` (using `npm link`)
- __TypeScript version__: `2.3.0-dev.20170323`
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
declare namespace N {
    let x: number;
}

N.x = 3;
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "prefer-const": true
    }
}
```

#### Actual behavior

`ERROR: a.ts[2, 9]: Identifier 'x' is never reassigned; use 'const' instead of 'let'.`

#### Expected behavior

No error. `prefer-const` should skip `.d.ts` files and anything marked `declare`.

**Comment by adidahiya** [comment]
Can you provide some more context / motivation around this?

Why shouldn't we lint .d.ts files? Rules generally don't have the responsibility of deciding which files they are enabled in

**Comment by andy-ms** [comment]
`prefer-const` detects whether something is mutated to determine whether it should be const.
In a declaration file, the variable will of course never be mutated, since it contains only declarations. It may be mutated by external code, but tslint has no way of knowing that.

**Comment by adidahiya** [comment]
ok, I buy that reasoning. makes sense
